www/codePush.ts

```diff
@@ -108,6 +115,119 @@ class CodePush implements CodePushCordovaPlugin {
+        };
+
+        var onDownloadSuccess = (localPackage: ILocalPackage) => {
+            localPackage.apply(onApplySuccess, onError, syncOptions.rollbackTimeout);
+        };
+
+        var downloadAndInstallUpdate = (remotePackage: RemotePackage) => {
+            remotePackage.download(onDownloadSuccess, onError);
+        };
+
+        var onUpdate = (remotePackage: RemotePackage) => {
+            if (!remotePackage || (remotePackage.failedApply && syncOptions.ignoreFailedUpdates)) {
+                syncCallback && syncCallback(SyncStatus.UP_TO_DATE);
+            } else {
+                if (remotePackage.isMandatory && syncOptions.mandatoryUpdateMessage) {
+                    /* Alert user */
+                    navigator.notification.alert(syncOptions.mandatoryUpdateMessage, () => { downloadAndInstallUpdate(remotePackage); }, syncOptions.dialogTitle, syncOptions.mandatoryUpdateContinueButtonLabel);
+                } else if (!remotePackage.isMandatory && syncOptions && syncOptions.optionalUpdateMessage) {
+                    /* Confirm update with user */
+                    var optionalUpdateCallback = (buttonIndex: number) => {
+                        switch (buttonIndex) {
+                            case 1:
+                                /* Install */
+                                downloadAndInstallUpdate(remotePackage);
+                                break;
+                            case 2:
+                            default:
+                                /* Cancel */
+                                syncCallback && syncCallback(SyncStatus.USER_DECLINED);
+                                break;
+                        }
+                    };
+
+                    navigator.notification.confirm(syncOptions.optionalUpdateMessage, optionalUpdateCallback, syncOptions.dialogTitle, [syncOptions.optionalUpdateConfirmButtonLabel, syncOptions.optionalUpdateCancelButtonLabel]);
+                } else {
+                    /* No user interaction */
+                    downloadAndInstallUpdate(remotePackage);
+                }
+            }
+        };
+
+        window.codePush.checkForUpdate(onUpdate, onError);
+    }
+
+    /**
+     * Returns the default options for the CodePush sync operation.
+     * If the options are not defined yet, the static DefaultSyncOptions member will be instantiated.
+     */
+    private getDefaultSyncOptions(): SyncOptions {
+        if (!CodePush.DefaultSyncOptions) {
+            CodePush.DefaultSyncOptions = {
+                dialogTitle: "Update",
+                mandatoryUpdateMessage: "You will be updated to the latest version.",
+                mandatoryUpdateContinueButtonLabel: "Continue",
+                optionalUpdateMessage: "An update is available. Would you like to install it?",
+                optionalUpdateConfirmButtonLabel: "Install",
+                optionalUpdateCancelButtonLabel: "Cancel",
```

Comment:
I was calling this "IgnoreButton" on the RN side, with the default text being "Ignore". I'm cool either way